pkg/sql/schemachanger/scdeps/sctestdeps/test_deps.go

```diff
@@ -906,6 +906,12 @@ func (s *TestState) SwapDescriptorSubComment(
 	return nil
 }
 
+// DeleteSchedule implements scexec.DescriptorMetadataUpdater
+func (s *TestState) DeleteSchedule(ctx context.Context, id int64) error {
+	s.LogSideEffectf("delete schedule ID %d", id)
```

Comment:
now that you've added this, can we get a nice data-driven test in `schemachanger/testdata`. You do, `setup` and statements to set up the world and then `test` for the drop of the table and let it rewrite.

Reply:
done. couldn't print out the schedule ID here though as it is non deterministic.